caas/kubernetes/clientconfig/k8s.go

```diff
@@ -15,15 +16,28 @@ import (
 
 var logger = loggo.GetLogger("juju.caas.kubernetes.clientconfig")
 
-// K8SClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.
-func K8SClientConfig() (*ClientConfig, error) {
+// NewClientConfig parses io.Reader to read the Kubernetes client configuration from
```

Comment:
// NewClientConfig returns a new Kubernetes client, reading the config from the specified reader.